services/blockchain-connector/shared/sdk/dynamicReward.js

```diff
@@ -43,26 +58,52 @@ const getRewardTokenID = async () => {
 	return rewardTokenID;
 };
 
+const getInflationRate = async () => {
+	try {
+		// TODO: Update endpoint once exposed by sdk
+		// Ref: https://github.com/LiskHQ/lisk-sdk/issues/7799
+		const inflationRate = await isModuleRegistered(MODULE.DYNAMIC_REWARD)
+			? await invokeEndpoint('dynamicReward_getInflationRate')
+			: await invokeEndpoint('reward_getInflationRate');
+
+		return inflationRate;
+	} catch (err) {
+		if (err.message.includes(timeoutMessage)) {
+			throw new TimeoutException('Request timed out when calling \'getInflationRate\'.');
+		}
+		const errMessage = await isModuleRegistered(MODULE.DYNAMIC_REWARD)
+			? `Error returned when invoking 'dynamicReward_getInflationRate'.\n${err.stack}`
+			: `Error returned when invoking 'reward_getInflationRate'.\n${err.stack}`;
+		logger.warn(errMessage);
```

Comment:
We can simplify it to something like below but `registeredRewardModule` could be determined at init and cached at the global level. The same could be applied to the part with endpoint invocation.

```suggestion
		const registeredRewardModule = await isModuleRegistered(MODULE.DYNAMIC_REWARD) ? 'dynamicReward' : 'reward';
		logger.warn(`Error returned when invoking '${registeredRewardModule}_getInflationRate'.\n${err.stack}`);
```